### PR TITLE
Dbsp angle case fix

### PR DESCRIPTION
**Pull request by eteq**
I don't really understand what might be different about the input files, but when I tried running PypeIt on a palomar DBSP dataset I have, this change was necessary to get it to run correctly. (it seems that the "D" in "deg" is sometimes capitalized and sometimes not...)

### CODE REVIEW

File: pypeit/spectrographs/p200_dbsp.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def compound_meta(self, headarr: List[fits.Header], meta_key: str):
             return Time(headarr[0]['UTSHUT']).mjd
         elif meta_key == 'dispangle':
             try:
-                return Angle(headarr[0]['ANGLE']).deg
+                return Angle(headarr[0]['ANGLE'].replace('Deg', 'deg')).deg
```
Comment (finagle29) [inline pypeit/spectrographs/p200_dbsp.py:98]: I think that using `headarr[0]['ANGLE'].lower()` might make the code more robust (for example if the DBSP controller writes `"20 Deg 37 Min"` in the `ANGLE` field).

**Review by finagle29 — CHANGES_REQUESTED** [review]
Thanks for finding this bug and contributing a fix for it!

File: pypeit/spectrographs/p200_dbsp.py (hunk 1 above)
Comment (jhennawi) [reply pypeit/spectrographs/p200_dbsp.py:98]: I agree, can someone make that change please, then I'll approve.

**Review by finagle29 — APPROVED** [review]

File: pypeit/spectrographs/p200_dbsp.py (hunk 1 above)
Comment (finagle29) [reply pypeit/spectrographs/p200_dbsp.py]: done. waiting for CI

**Review by jhennawi — APPROVED** [review]
Approved.

File: pypeit/spectrographs/p200_dbsp.py (hunk 1 above)
Comment (eteq) [reply pypeit/spectrographs/p200_dbsp.py]: Oops, missed this thread until now, but thanks for taking care of it, @finagle29 !

**Pull request merged**